packages/core/src/feed.ts

```diff
@@ -1,39 +1,68 @@
 import { EventType, StreamMetadata, StreamState } from '@ceramicnetwork/common'
-import { Subject, type Observable } from 'rxjs'
-import { CommitID } from '@ceramicnetwork/streamid'
+import { CommitID, StreamID } from '@ceramicnetwork/streamid'
 import { StreamUtils } from '@ceramicnetwork/common'
+import type { FeedAggregationStore, AggregationStoreEntry } from './store/feed-aggregation-store.js'
 
 export class FeedDocument {
   constructor(
+    readonly resumeToken: string,
     readonly commitId: CommitID,
     readonly content: any,
     readonly metadata: StreamMetadata,
     readonly eventType: EventType
   ) {}
 
-  static fromStreamState(streamState: StreamState): FeedDocument {
-    return {
-      commitId: StreamUtils.commitIdFromStreamState(streamState),
-      content: streamState.next ? streamState.next.content : streamState.content,
-      metadata: streamState.next ? streamState.next.metadata : streamState.metadata,
-      eventType: streamState.log[streamState.log.length - 1].type,
-    }
+  static fromStreamState(resumeToken: string, streamState: StreamState): FeedDocument {
+    return new FeedDocument(
+      resumeToken,
+      StreamUtils.commitIdFromStreamState(streamState),
+      streamState.next ? streamState.next.content : streamState.content,
+      streamState.next ? streamState.next.metadata : streamState.metadata,
+      streamState.log[streamState.log.length - 1].type
+    )
   }
 }
 
-class DocumentsSubject extends Subject<FeedDocument> {}
-
 /**
  * Read-only version of `Feed`.
  */
 export interface PublicFeed {
   aggregation: {
-    documents: Observable<FeedDocument>
+    documents: (after?: string) => ReadableStream<FeedDocument>
   }
 }
 
 export class Feed implements PublicFeed {
-  readonly aggregation = {
-    documents: new DocumentsSubject(),
+  constructor(
+    private readonly feedStore: FeedAggregationStore,
+    private readonly streamState: (streamId: StreamID) => Promise<StreamState | undefined>
+  ) {}
+
+  get aggregation() {
+    return {
+      documents: this.documents.bind(this),
+    }
+  }
+
+  documents(after?: string): ReadableStream<FeedDocument> {
+    const transformer = new StreamLoadTransformer(this.streamState)
+    return this.feedStore.streamIDs(after).pipeThrough(new TransformStream(transformer))
+  }
+}
+
+export class StreamLoadTransformer implements Transformer<AggregationStoreEntry, FeedDocument> {
+  constructor(
+    private readonly streamState: (streamId: StreamID) => Promise<StreamState | undefined>
+  ) {}
+
+  async transform(
+    entry: AggregationStoreEntry,
+    controller: TransformStreamDefaultController<FeedDocument>
+  ) {
+    const found = await this.streamState(entry.streamID)
+    if (found) {
```

Comment:
what would it mean if we didn't find the stream state here?  Seems like that probably shouldn't ever happen right?  Should we at least log a warning or something?